src/app/css-variables.ts

```diff
@@ -1,56 +1,74 @@
 import { settingsSelector } from 'app/dim-api/selectors';
+import { deepEqual } from 'fast-equals';
 import { isPhonePortraitSelector } from './shell/selectors';
-import { observeStore } from './utils/redux';
+import { StoreObserver } from './store/observerMiddleware';
 
 function setCSSVariable(property: string, value: { toString: () => string }) {
   if (value) {
     document.querySelector('html')!.style.setProperty(property, value.toString());
   }
 }
 
+export function createItemSizeObserver(): StoreObserver<number> {
+  return {
+    id: 'item-size-observer',
+    getObserved: (rs) => settingsSelector(rs).itemSize,
+    sideEffect: ({ previous, current }) => {
+      if (previous !== current) {
+        setCSSVariable('--item-size', `${Math.max(48, current)}px`);
+      }
+    },
+  };
+}
+
+export function createThemeObserver(): StoreObserver<{ theme: string; isPhonePortrait: boolean }> {
+  return {
+    id: 'theme-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      theme: settingsSelector(rs).theme,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ previous, current }) => {
+      // TODO (ryan): Hoist the feature flag check up to where the observer is created
+      // Set a class on the body to control the theme. This must be applied on the body for syncThemeColor to work.
+      if ($featureFlags.themePicker && previous.theme !== current.theme) {
```

Comment:
Either this also needs to check if `isPhonePortrait` is changed, or `syncThemeColor` doesn't actually depend on `isPhonePortrait`.

Reply:
I will remove it, I don't think it should be in the same observer and instead be handled entirely in the theme or isPhonePortrait specific handlers. 

It's largely there because it was there before and I didn't dig into why it was there.

Reply:
@bhollis `isPhonePortrait` is still necessary for `syncColorTheme` as it pulls the background color from different variables. See https://github.com/DestinyItemManager/DIM/blob/91cde0d71f488606b65ab1bd3a4b5f79fc7f9749/src/app/css-variables.ts#L83-L97


I have just removed all the checks as you will see and hoisted the flag check up above setting up the observer. I don't think its worth guarding against noise in changes as the observer middleware checks for equality.